firebase/FirestoreCtrl.ts

```diff
@@ -180,4 +180,28 @@ export default class FirestoreCtrl {
       throw error;
     }
   }
+
+  /**
+   * Retrieves all challenges from Firestore. WARNING: This can be slow and expensive.
+   *
+   * @returns A promise that resolves to an array of all challenges.
+   */
```

Comment:
This can easily be overwhelmed if the collection is really large. I guess it's fine for now but maybe in a future PR we could improve this by using another algorithm. For now we can just limit the number of challenges retrieved by using limit in query.

Reply:
Totally agreed with you, I'll see if it's technically positive to introduce a query limit.